common/types/verified_activation.go

```diff
@@ -0,0 +1,69 @@
+package types
+
+import "github.com/spacemeshos/go-spacemesh/log"
+
+type VerifiedActivationTx struct {
+	*ActivationTx
+
+	baseTickHeight uint64
+	tickCount      uint64
+}
+
+// GetWeight of the atx.
+func (vatx *VerifiedActivationTx) GetWeight() uint64 {
+	return uint64(vatx.NumUnits) * (vatx.tickCount)
+}
+
+// BaseTickHeight is a tick height of the positional atx.
+func (vatx *VerifiedActivationTx) BaseTickHeight() uint64 {
+	return vatx.baseTickHeight
+}
+
+// TickCount returns tick count from from poet proof attached to the atx.
+func (vatx *VerifiedActivationTx) TickCount() uint64 {
+	return vatx.tickCount
+}
+
+// TickHeight returns a sum of base tick height and tick count.
+func (vatx *VerifiedActivationTx) TickHeight() uint64 {
+	return vatx.baseTickHeight + vatx.tickCount
+}
+
+// Header returns the header of the atx.
+func (vatx *VerifiedActivationTx) Header() *ActivationTxHeader {
```

Comment:
What's the motivation to return a pointer here instead of just a value? The users of this API might think that this method returns a pointer to the insides of `vatx`, but it returns a separate object (a copy basically). And they must consider what happens if `nil` is returned (I know it cannot return a `nil` but to know that you need to read the implementation).

I'd suggest returning by value. What do you think?

Reply:
my understanding was that headers are always cached, hence the ptr

Reply:
The change I made is instead of calling `header := &atx.ActivationTxHeader` one now calls `header := atx.Header()` unless someone explicitly fetches the ATX or its header from the store it will NOT be cached and wasn't cached before.